internal/baggage/restriction_manager.go

```diff
@@ -18,32 +18,51 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-package jaeger
+package baggage
 
 const (
 	defaultMaxValueLength = 2048
 )
 
-// baggageRestrictionManager keeps track of valid baggage keys and their size restrictions.
-type baggageRestrictionManager interface {
-	// setBaggage sets the baggage key:value on a span and the associated logs.
-	setBaggage(span *Span, key, value string)
+type Restriction struct {
```

Comment:
I tried to make this immutable by adding the getters. This way we only need to generate this once and return it to the setter